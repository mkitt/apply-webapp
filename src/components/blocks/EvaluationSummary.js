// @flow
import React from 'react'
import glamorous from 'glamorous'
import TextLink from '../elements/TextLink'
import { COLORS } from '../../constants/theme'

type Props = {
  score?: number | null,
  term: string,
  to?: string | null,
}

const hasScore = score => (
  score != null && score > 0
)

const EvaluationSummary = ({ score, term, to, ...props }: Props) => (
  <View {...props}>
    <Heading>
      <TermText>{term}</TermText>
      {hasScore(score) &&
        <ScoreText>{score}</ScoreText>
      }
    </Heading>
    { to != null &&
      <ScoreLink to={to}>
        {`${hasScore(score) ? 'Re-' : ''}Score Evaluation`}
      </ScoreLink>
    }
  </View>
)

EvaluationSummary.defaultProps = {
  score: null,
  to: null,
}

export default EvaluationSummary

// -------------------------------------

const View = glamorous.div({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  marginTop: 44,
  borderBottom: `1px solid ${COLORS.GREY_C}`,
})

const Heading = glamorous.h2({
  display: 'flex',
  flexFlow: 'row nowrap',
  fontSize: 28,
  fontWeight: 300,
  color: COLORS.GREY_8,
})

const TermText = glamorous.span()

const ScoreText = glamorous.span({
  color: COLORS.LIGHT_GREEN,
  marginLeft: 8,
  '&::before': {
    content: ' : ',
    color: COLORS.GREY_8,
  },
})

const ScoreLink = glamorous(TextLink)({
  fontSize: 18,
})
