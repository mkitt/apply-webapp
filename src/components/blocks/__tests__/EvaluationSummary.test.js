import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import EvaluationSummary from '../EvaluationSummary'

describe('<EvaluationSummary />', () => {
  it('shows the correct snapshot tree for the <EvaluationSummary> block', () => {
    const component = (
      <MemoryRouter>
        <EvaluationSummary
          score={100}
          term="Evaluation Summary Term"
          to="/to-evaluation-summary"
        />
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
