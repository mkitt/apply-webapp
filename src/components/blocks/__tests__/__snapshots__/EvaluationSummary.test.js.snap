// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EvaluationSummary /> shows the correct snapshot tree for the <EvaluationSummary> block 1`] = `
.glamor-4,
[data-glamor-4] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 44px;
  border-bottom: 1px solid #ccc;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row nowrap;
  font-size: 28px;
  font-weight: 300;
  color: #808080;
  -webkit-flex-flow: row nowrap;
}

.glamor-1,
[data-glamor-1] {
  color: #8ad30e;
  margin-left: 8px;
}

.glamor-1::before,
[data-glamor-1]::before {
  content: " : ";
  color: #808080;
}

.glamor-3,
[data-glamor-3] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  font-size: 18px;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<div
  className="glamor-4"
>
  <h2
    className="glamor-2"
  >
    <span
      className="glamor-0"
    >
      Evaluation Summary Term
    </span>
    <span
      className="glamor-1"
    >
      100
    </span>
  </h2>
  <a
    className="glamor-3"
    fontSize={18}
    href="/to-evaluation-summary"
    onClick={[Function]}
  >
    Re-Score Evaluation
  </a>
</div>
`;
